posix-dirent-name.go

```diff
@@ -0,0 +1,36 @@
+// +build linux nacl solaris
+
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+func direntIno(buf []byte) (uint64, bool) {
+	return readInt(buf, unsafe.Offsetof(syscall.Dirent{}.Ino), unsafe.Sizeof(syscall.Dirent{}.Ino))
+}
+
+func direntNamlen(buf []byte) (uint64, bool) {
+	reclen, ok := direntReclen(buf)
+	if !ok {
+		return 0, false
+	}
+	return reclen - uint64(unsafe.Offsetof(syscall.Dirent{}.Name)), true
```

Comment:
i did:
`fmt.Println(uintptr(dirent.Reclen)-unsafe.Offsetof(syscall.Dirent{}.Name), clen(bytes[:]), name)` in https://gist.github.com/krishnasrinivas/ee4caa17dbe6af00eb17a5af10182d3d

it prints:
```
krishna@escape:~/tmp$ go run ~/posix-list-dir-nix.go
13 7 sort.go
13 12 node_modules
13 8 testlist
13 10 readdir.go
```

i.e clen() prints the correct length, `reclen - uint64(unsafe.Offsetof(syscall.Dirent{}.Name))` does not

if you print reclen, it is always a multiple of 8 indicating that it accounts for padding, hence checking for null termination (how clen() does) is reliable.